clients/drcachesim/reader/compressed_file_reader.cpp

```diff
@@ -52,7 +52,13 @@ namespace drmemtrace {
 bool
 open_single_file_common(const std::string &path, gzFile &out)
 {
-    out = gzopen(path.c_str(), "rb");
+    if (path == "-") {
+        // We assume stdin is 0.
+        // We do not use "stdin->_fileno" as that is not portable.
```

Comment:
Can you clarify what this is trying to say? What is "stdin->_fileno"?

Reply:
glibc's `stdin` is a FILE* and its file descriptor is `stdin->_fileno` and In other DR code we hardcode this: yet musl libc does not have that `_fileno` field, if you noticed as part of the recent musl PR: so it's a way we've obtained the stdin number in other DR code, but I'm pointing out it's not portable so we shouldn't do it.

Rewrote as:
```
        // We do not use the glibc-specific "stdin->_fileno" method of finding the
        // stdin descriptor number as that is not portable to musl or other libc.
```